lib/cookie/cookieJar.ts

```diff
@@ -186,36 +188,42 @@ export class CookieJar {
     return syncResult
   }
 
+  // TODO: We *could* add overloads based on the value of `options.ignoreError`, such that we only
+  // return `undefined` when `ignoreError` is true. But would that be excessive overloading?
   setCookie(
     cookie: string | Cookie,
     url: string,
-    callback: Callback<Cookie>,
+    callback: Callback<Cookie | undefined>,
   ): void
   setCookie(
     cookie: string | Cookie,
     url: string,
     options: SetCookieOptions,
-    callback: Callback<Cookie>,
+    callback: Callback<Cookie | undefined>,
   ): void
-  setCookie(cookie: string | Cookie, url: string): Promise<Cookie>
+  setCookie(cookie: string | Cookie, url: string): Promise<Cookie | undefined>
   setCookie(
     cookie: string | Cookie,
     url: string,
     options: SetCookieOptions,
-  ): Promise<Cookie>
+  ): Promise<Cookie | undefined>
   setCookie(
     cookie: string | Cookie,
     url: string,
-    options: SetCookieOptions | Callback<Cookie>,
-    callback?: Callback<Cookie>,
+    options: SetCookieOptions | Callback<Cookie | undefined>,
+    callback?: Callback<Cookie | undefined>,
   ): unknown
   setCookie(
     cookie: string | Cookie,
     url: string,
-    options?: SetCookieOptions | Callback<Cookie>,
-    callback?: Callback<Cookie>,
+    options?: SetCookieOptions | Callback<Cookie | undefined>,
+    callback?: Callback<Cookie | undefined>,
   ): unknown {
-    const promiseCallback = createPromiseCallback<Cookie>(arguments)
+    if (typeof options === 'function') {
+      callback = options
+      options = undefined
+    }
```

Comment:
Since we have mushed up function signatures, we have to rearrange the parameters to use them correctly.